Source/Controllers/Preferences/Panes/SPFilePreferencePane.m

```diff
@@ -480,37 +480,45 @@ - (void)chooseFileWithOptions:(PanelOptions*)options
         // read-only.
         [self->_currentFilePanel.URLs enumerateObjectsUsingBlock:^(NSURL *url, NSUInteger idxURL, BOOL *stopURL){
 
-            if([SecureBookmarkManager.sharedInstance addBookmarkForUrl:self->_currentFilePanel.URL options:(NSURLBookmarkCreationWithSecurityScope|NSURLBookmarkCreationSecurityScopeAllowOnlyReadAccess) isForStaleBookmark:options.isForStaleBookmark] == YES){
-                SPLog(@"addBookmarkForUrl success");
-                CLS_LOG(@"addBookmarkForUrl success");
-
-                if(options.isForStaleBookmark == YES){
-
-                    // Here we need to maintain the options filnames array
-                    // and the selectedRows index
-                    SPLog(@"options.fileNames: %@", options.fileNames);
-                    SPLog(@"self->selectedRows: %@", self->selectedRows);
-                    SPLog(@"removing stale file from options.fileNames at index 0");
-                    SPLog(@"removing stale file from self->selectedRows at index: %lu", (unsigned long)options.index);
-                    CLS_LOG(@"removing stale file from options.fileNames");
-
-                    SPLog(@"selectedRows count = %lu", (unsigned long)self->selectedRows.count);
-
-                    // we need to keep track of how many files there are to prompt for
-                    // so we remove from the filenames array. Index 0 is safe.
-                    [options.fileNames removeObjectAtIndex:0];
-                    // to keep things nice and tidy, remove the index?
-                    [self->selectedRows removeIndex:options.index];
-                    SPLog(@"options.fileNames: %@", options.fileNames);
-                    SPLog(@"self->selectedRows: %@", self->selectedRows);
-
-                    // increment for the next file
-                    options.index++;
-                }
+            // check it's really a URL
+            if(![self->_currentFilePanel.URL isKindOfClass:[NSURL class]]){
+                SPLog(@"self->keySelectionPanel.URL is not a URL: %@", self->_currentFilePanel.URL.class);
+                CLS_LOG(@"self->keySelectionPanel.URL is not a URL: %@", self->_currentFilePanel.URL.class);
+                // JCS - should we stop here?
```

Comment:
Probably?

Reply:
reviewing at the moment...

Reply:
Ah, just the act of selecting the file in the NSOpenPanel calls `startAccessingSecurityScopedResource`

I swizzled `startAccessingSecurityScopedResource`. This is output as you choose the file in the export window:

```log
2021-01-15 19:18:43.301411+0800 Sequel Ace[54819:3312141] -[NSURL(SPURLAdditions)
 SA_startAccessingSecurityScopedResource]:36: JIMMY startAccessingSecurityScopedResource called:
 file:///Users/james/untitled%20folder%202/untitled%20folder/untitled%20folder/
```

So the only downside of this failing is that we won't have a bookmark, and therefore no access on the next app start. I don't think we should stop.

Reply:
Oh, hold on I had two places where I put "JCS - should we stop here?" ... 

1. Not a valid URL
2. Couldn't create a bookmark.

I was talking about 2 above...

Reply:
But still .. .if it's not a URL, we don't try to create a bookmark ... but then how does the exporter (or whatever) read/write to the file? Need an example of an invalid URL....

Reply:
so the logging I added in this PR should give us some examples.